src/theory/booleans/circuit_propagator_proofs.h

```diff
@@ -0,0 +1,580 @@
+/*********************                                                        */
+/*! \file circuit_propagator_proofs.h
```

Comment:
I'd call this file proof_circuit_propagator.h / cpp to follow our general naming conventions of proof-producing classes.